README.md

```diff
@@ -0,0 +1,81 @@
+# Checkly Terraform provider
+
+[![CircleCI](https://circleci.com/gh/tnolet/terraform-provider-checkly/tree/master.svg?style=svg)](https://circleci.com/gh/tnolet/terraform-provider-checkly/tree/master)
+
+This Terraform provider enables users to manage [Checkly](https://checklyhq.com) resources like checks.
+
+## Using the provider
+
+1. Download the binary applicable for your platform from the [latest tagged release](https://github.com/tnolet/terraform-provider-checkly/releases)
+
+```bash
+
+curl -OL https://github.com/tnolet/terraform-provider-checkly/releases/latest/download/terraform-provider-checkly_darwin_amd64.gz
+```
+
+2. Copy the binary to your Terraform plugin folder, unzip it and rename it to just `terraform-provider-checkly`
+
+```bash
+cp terraform-provider-checkly_darwin_amd64.gz ~/.terraform.d/plugins/darwin_amd64
+gunzip ~/.terraform.d/plugins/darwin_amd64/terraform-provider-checkly_darwin_amd64.gz
+mv ~/.terraform.d/plugins/darwin_amd64/terraform-provider-checkly_darwin_amd64 ~/.terraform.d/plugins/darwin_amd64/terraform-provider-checkly
+```
+
+3. Run `terraform init` and then `terraform providers`. The Checkly plugin should be listed.
+
+```bash
+terraform init
+
+Initializing provider plugins...
+Terraform has been successfully initialized!
+
+terraform providers
+.
+└── provider.checkly
+```
+
+If you're having issues, please check [the Hashicorp docs on installing third party plugins.](https://www.terraform.io/docs/configuration/providers.html#third-party-plugins)
+
+### Usage examples
+
+Add a `checkly_check` resource to your resource file
+
+```terraform
+resource "checkly_check" "checkly-public-stats" {
```

Comment:
A more detailed example would be helpful here. In fact, what we could do is supply a 'minimal' example—just the basics to get a simple endpoint check working—and additionally a more featureful one which shows off everything Checkly does. For example, from the `test.tf` file:

```terraform
variable "checkly_api_key" {}

provider "checkly" {
  api_key = "${var.checkly_api_key}"
}

resource "checkly_check" "test-check2" {
  name             = "My test check 2"
  type             = "API"
  activated        = true
  should_fail      = true
  frequency        = 1
  ssl_check_domain = "example.com"
  double_check     = true

  locations = [
    "us-west-1",
    "ap-northeast-1",
    "ap-south-1",
  ]

  alert_settings {
    ssl_certificates {
      enabled         = true
      alert_threshold = 30
    }
  }

  request {
    follow_redirects = true
    url              = "http://example.com/"

    query_parameters = {
      search = "foo"
    }

    headers = {
      X-Bogus = "bogus"
    }

    assertion {
      source     = "JSON_BODY"
      property   = "code"
      comparison = "HAS_VALUE"
      target     = "authentication.failed"
    }

    assertion {
      source     = "STATUS_CODE"
      comparison = "EQUALS"
      target     = "401"
    }
  }
}
```

Reply:
I was looking for ways to generate documentation for TF providers, but didn't find anything. That is kind of unbelievable. There must be away to generate this from Go docs or something?

Reply:
You'd think!

Reply:
ha ha, well. Let me just resolve this one for now and I will dig up some nice examples and create a new PR just for that.